uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/common/extensions/SdkExtensions.kt

```diff
@@ -173,4 +185,93 @@ fun Application.computePrimaryRoute(routePlan: RoutePlan, routeComputeCallback:
+    this@concatItems.forEach {
+        if (isNotEmpty()) append(", ")
+        append(it.text)
+    }
+}.toString()
+
+fun List<NaviSignInfo>.getNaviSignInfoOnRoute(): NaviSignInfo? {
+    this.filter { it.isOnRoute }.let { isOnRouteList ->
+        if (isOnRouteList.isEmpty()) {
+            return null
+        }
+
+        //ToDo: Use NaviSignInfo "priority" when ready
+        return isOnRouteList.firstOrNull { it.backgroundColor != 0 }?.let { it } ?: isOnRouteList[0]
+    }
+}
+
+fun NaviSignInfo.roadSigns(maxRoadSignsCount: Int = 3): List<RoadSignData> {
+    return signElements
+        .asSequence()
+        .filter { it.elementType == SignElementType.RouteNumber }
+        .filter { it.routeNumberFormat.insideNumber.isNotEmpty() }
+        .take(if (signElements.hasPictogram()) maxRoadSignsCount - 1 else maxRoadSignsCount)
+        .toList()
+        .toRoadSignDataList()
+}
+
+fun NaviSignInfo.createInstructionText(): TextHolder {
+    val acceptedSignElements = signElements
+        .filter { element ->
+            element.elementType.let {
+                it == SignElementType.ExitNumber || it == SignElementType.PlaceName || it == SignElementType.OtherDestination
+            }
+        }
+        .sortedByDescending { it.elementType == SignElementType.ExitNumber }
+
+    return if (acceptedSignElements.any { it.elementType == SignElementType.ExitNumber }) {
+        TextHolder.from(R.string.exit_number, acceptedSignElements.concatItems())
+    } else {
+        TextHolder.from(acceptedSignElements.concatItems())
+    }
+}
+
+fun List<NaviSignInfo.SignElement>.hasPictogram(): Boolean = any { it.elementType == SignElementType.Pictogram }
+
+fun List<NaviSignInfo.SignElement>.toRoadSignDataList(): List<RoadSignData> {
+    return map {
+        with(it.routeNumberFormat) {
+            RoadSignData(roadSignBackgroundDrawableRes(), insideNumber, roadSignForegroundColorRes())
+        }
+    }
+}
+
+fun DirectionInfo.getDistanceWithUnits(distanceUnit: DistanceUnit): String =
+    Distance.getFormattedDistance(distanceUnit, distance)
+
+@DrawableRes
+fun DirectionInfo.getDirectionDrawable(directionManeuverType: DirectionManeuverType): Int {
+    val routeManeuver = when (directionManeuverType) {
+        DirectionManeuverType.PRIMARY -> primary
+        DirectionManeuverType.SECONDARY -> secondary
+    }
+
+    return if (routeManeuver.isValid) routeManeuver.getDirectionDrawable() else 0
+}
+
+fun DirectionInfo.createInstructionText(): TextHolder {
+    with(primary) {
+        if (!isValid) {
+            return TextHolder.empty
+        }
+
+        if (isRoundabout()) {
+            getDirectionInstruction().let {
+                return if (it != 0) TextHolder.from(it, roundaboutExit) else TextHolder.empty
+            }
+        }
+
+        with(nextRoadText()) {
+            if (isNotEmpty()) return TextHolder.from(this)
+        }
+
+        getDirectionInstruction().let {
```

Comment:
nebolo by to lepsie: 
```suggestion
        return getDirectionInstruction().let {
```